Fix SvelteKit example

DePasqualeOrg opened the pull request:
The SvelteKit example was broken and wrong in a number of ways. It could still be cleaned up, but at least it builds now.

DePasqualeOrg commented on examples/svelte-kit-example/src/routes/sign-up/+page.svelte:
```diff
@@ -1,33 +1,35 @@
-	});
-};
+  // Function to handle form submission
+  const handleSignUp = async () => {
```
AI slop

DePasqualeOrg commented on examples/svelte-kit-example/src/routes/sign-up/+page.svelte:
```diff
@@ -66,12 +68,12 @@
       <div class="grid gap-2">
         <Label for="password">Password</Label>
-        <Input id="password" type="password" bind:value={$password} />
+        <Input id="password" type="password" bind:value={password} />
```
AI slop

DePasqualeOrg commented:
@Bekacru @Kinfe123, your SvelteKit example is completely broken, and some of the code examples are not correct. I spent a lot of time fixing this for you, and I just resolved some merge conflicts that emerged after I submitted this PR for review. Could you please review and merge this before more conflicts emerge?

Kinfe123 commented:
@DePasqualeOrg will look into it.

vladshcherbin commented:
Why does this example need a hundred of UI components and packages? Just show some basic UI and actual auth setup without bloat.

It's insane 🤯

DePasqualeOrg commented:
It doesn't need most them. It only uses a few, so I removed all the unused ones. I also don't understand how a completely broken example that doesn't even build got pushed to production.

DePasqualeOrg commented:
@Bekacru @Kinfe123, I submitted this fix more than 6 weeks ago, and you have ignored it. It's a pretty big red flag to me that something like this got merged in the first place, and that the fix simply gets ignored.

DePasqualeOrg commented:
@Bekacru @Kinfe123, over three months after I submitted this fix, your SvelteKit example is still completely broken.

DePasqualeOrg commented:
@Bekacru @Kinfe123, I see that you merged https://github.com/better-auth/better-auth/pull/1742 without engaging here. However, the SvelteKit example is still broken due to the issues with Biome that I mentioned above. I have rebased this PR and fixed those issues here, so that the example now builds and runs. However, I noticed that there are still issues with the sign-up flow. I would appreciate it if you could finally address this PR.

Bekacru commented:
@DePasqualeOrg sorry about leaving you hanging here. Can you please fix the CI?

DePasqualeOrg commented:
It looks like it's failing because of a frozen lockfile. How can this be fixed?

Bekacru commented:
re-install locally and make sure to commit the lock file

DePasqualeOrg commented:
The CI tests are now passing.

Bekacru commented on examples/svelte-kit-example/.prettierrc:
```diff
@@ -0,0 +1,16 @@
+{
```
we wouldn't want to introduce prettier on top of biome

Bekacru commented:
There should be no reason to remove the shadcn components. We want to have all of them there so it'd be easier for testing

Bekacru commented:
I just tried building the current SvelteKit example, and it seems to build fine. Biome also appears to format it properly. Could you let me know exactly where it's broken?

Bekacru commented:
> @Bekacru, what do you mean having extra, unused components makes testing easier? On the contrary, I don't see any reason to keep this dead code. If you want to keep unnecessary components, why just these ones and not all the shadcn-svelte components from the registry? It doesn't make sense to me.

these are every component from the svelte shadcn registry if I recall properly :)

Bekacru commented on examples/svelte-kit-example/eslint.config.js:
```diff
@@ -0,0 +1,42 @@
+import prettier from 'eslint-config-prettier';
```
yes. Probably at this point if the example is such an issue and there is no way to fix through Biome, we'd probably move it to a different repo to avoid introducing another linter

Bekacru replied:
yes

DePasqualeOrg commented:
> I just tried building the current SvelteKit example, and it seems to build fine. Biome also appears to format it properly. Could you let me know exactly where it's broken?

Biome doesn't work with SvelteKit and was stripping out required code, which is why the SvelteKit example was broken. The only reason it works now is because I fixed it by making Biome ignore the SvelteKit example.

Bekacru commented:
I'm saying, the one in the main branch works, Biome isn't stripping any code. I'm not sure what is broken about it.

Bekacru commented:
Anyway, I feel like you're not engaging in good faith. Thanks for flagging this, we'll improve the SvelteKit example soon and if there's anything to fix, we'll address it then. In the meantime, there are several better-auth SvelteKit examples out there. Feel free to check those out. This one (like some of the others examples) hasn't been updated recently, but hopefully all of them will be soon.

DePasqualeOrg commented:
@Bekacru, what are you talking about? You're the one who's not engaging in good faith by closing this PR, which fixes a legitimate issue.

Your claim that the SvelteKit example already works is not true: I just cloned the main branch and verified that the SvelteKit example does not build or run in dev mode. Run `pnpm run check` in the SvelteKit example and you will see all the missing code that has already been stripped by Biome.

You gave a reason for keeping dead code that doesn't make sense, and then instead of engaging in a rational discussion, you simply closed this PR.

To anyone else watching this PR, what do you think about this behavior?
